### DISCUSSION TITLE
RTF text in OBX

### DISCUSSION THREAD

**Discussion by tmartinac** [nextgenhealthcare/connect#5691] [Q&A] [answered]
I am trying to take a standard ORU message inbound and send out an ORU message outbound with essentially a RTF file in text format in OBX.5

There will be fields from the inbound ORU that need to go into the RTF, as well as the text from the OBX segments.

I've tried to build out the RTF using a string variable, but Mirth is not happy with some of the text of the RTF (in the editor it turns my text color from pink to gray)

Any thoughts on how to better handle this request?

**Comment by jonbartels** [comment]
> I've tried to build out the RTF using a string variable, but Mirth is not happy with some of the text of the RTF (in the editor it turns my text color from pink to gray)

This is not necessarily a fatal problem. The MC syntax highlighter will occasionally mark some text as bad when more robust parsers are fine with it. Do you get the same problem in HL7Inspector or similar tools?

Are you base64 encoding your RTF or sending it raw?

You need to show a sample message and screenshots to get help.

**Reply by tmartinac** [reply]
I've attached the message type I am trying to create on my outbound transformer (result_243.txt)

It appears Mirth is parsing the code and it does not like '/f', which it is converting to &#12 and is giving an error:

Character reference "&#12" is an invalid XML character.

Mirth Connect 4.0.1
[result_243.txt](https://github.com/nextgenhealthcare/connect/files/10796575/result_243.txt)
[khs_SampleMirthCode.txt](https://github.com/nextgenhealthcare/connect/files/10796579/khs_SampleMirthCode.txt)

**Reply by pacmano1** [reply]
I suspect you need to escape the entire RTF string.  Code below from githib or forums, I forget.

```JS
function formatHL7String(str)
{
	var hl7String = str.replace(/\\/g, '\\E\\');
	hl7String = hl7String.replace(/\|/g,'\\F\\');
	hl7String = hl7String.replace(/\^/g,'\\S\\');
	hl7String = hl7String.replace(/\~/g,'\\R\\');
	hl7String = hl7String.replace(/\&/g,'\\T\\');
	return hl7String;
}
```

**Comment by kirbykn2** [comment]
Is this causing a problem?

On Tue, Feb 21, 2023 at 1:38 PM tmartinac ***@***.***> wrote:

> I've attached the message type I am trying to create on my outbound
> transformer (result_243.txt)
>
> It appears Mirth is parsing the code and it does not like '/f', which it
> is converting to &#12
> <https://github.com/nextgenhealthcare/connect/issues/12> and is giving an
> error:
>
> Character reference "&#12
> <https://github.com/nextgenhealthcare/connect/issues/12>" is an invalid
> XML character.
>
> Mirth Connect 4.0.1
> result_243.txt
> <https://github.com/nextgenhealthcare/connect/files/10796575/result_243.txt>
> khs_SampleMirthCode.txt
> <https://github.com/nextgenhealthcare/connect/files/10796579/khs_SampleMirthCode.txt>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/5691#discussioncomment-5068145>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD27CCJWVRCNCBWKI6LWYUDSTANCNFSM6AAAAAAVCL7OUY>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <nextgenhealthcare/connect/repo-discussions/5691/comments/5068145@
> github.com>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Comment by tmartinac** [comment, marked as answer]
Thank you to everyone for your suggestions for my issue. 

In the end, I was not using double-backslashes for my single backslashes.  Once I add those extras the errors went away and I was able to produce the RTF text I needed for my OBX.5-1 segment.

**Reply by pacmano1** [reply]
That's what the function above does.   You are supposed to escape a \ with `\E\`
